integration-cli/docker_cli_build_test.go

```diff
@@ -2247,7 +2247,7 @@ func (s *DockerSuite) TestBuildCmd(c *check.C) {
 
 func (s *DockerSuite) TestBuildExpose(c *check.C) {
 	name := "testbuildexpose"
-	expected := "map[2375/tcp:{}]"
+	expected := "map[2375/tcp:map[]]"
```

Comment:
I believe this is being shown as an empty map rather than as an empty struct because we're only looking at the raw JSON. According to the golang docs on JSON Unmarshal, generic JSON objects (like {}) will be converted to a map - which is what we're seeing above.

Not sure yet which way to go on this. I see two options:
1 - leave it like this and do some hand waving about how Go deals with JSON serialization.
2 - modify the API types along the lines of what was discussed above. Meaning, make the API types match exactly what we want to see on the wire and what the template stuff should produce.   So, @shishir-a412ed when you first mentioned modifying the data structures, that may be right approach, but it would need to only be done for the externalized data structures and not the internal ones.  Then we can deserialize into the real types and use those in the template stuff, and not the raw json.

Unless people think showing 'map' is correct here?  Feels wrong though.

Need to think about this some more... but I'm thinking something more drastic like option 2 might be the right approach.